src/components/ListItem.tsx

```diff
@@ -0,0 +1,108 @@
+import { Transition } from "@headlessui/react";
+import Image from "next/image";
+import { useState } from "react";
+import { Tick } from "./Tick";
+import {
+  LeadingActions,
+  SwipeableList,
+  SwipeableListItem,
+  SwipeAction,
+  TrailingActions,
+} from "react-swipeable-list";
+import "react-swipeable-list/dist/styles.css";
+
+type ListItemProps = {
+  name: string;
+  image: string;
+  quantity?: string;
+  checked: boolean;
+  onRemove: () => void;
+};
+// onRemove is added so higher level page can trigger a function when an item is removed.
+export default function ListItem(props: ListItemProps) {
+  const [checked, setChecked] = useState(false);
+  const [swiped, setSwiped] = useState(true);
+  const onCheck = () => {
+    setChecked(!checked);
+  };
+  const onDelete = () => {
+    setSwiped(false);
+    props.onRemove();
+  };
+  const leadingActions = () => (
+    <LeadingActions>
+      <SwipeAction onClick={onCheck} Tag="div">
+        <div
+          id="Tick_Action"
+          className="w-full h-full bg-ux-success button-bold text-text-white flex items-center"
+        >
+          Tick
+        </div>
+      </SwipeAction>
+    </LeadingActions>
+  );
+  const trailingActions = () => (
+    <TrailingActions>
+      <SwipeAction onClick={onDelete} Tag="div">
+        <div
+          id="Delete_Action"
+          className=" bg-ux-error button-bold text-text-white flex items-center justify-center text-left"
+        >
+          <p>Delete</p>
+        </div>
+      </SwipeAction>
+    </TrailingActions>
+  );
+
+  return (
+    <>
+      <Transition
+        show={swiped}
+        appear={true}
+        enter="transform transition duration-[400ms]"
+        enterFrom="opacity-0 scale-y-0"
+        enterTo="opacity-100 scale-y-100"
+        leave="transform duration-[400ms] transition ease-in-out"
+        leaveFrom="opacity-100 scale-y-100 "
+        leaveTo="opacity-0 scale-y-0 "
+      >
+        <SwipeableListItem
+          className="max-w-[334px] h-16 bg-primary-transparent border-[1px] border-secondary-default rounded-md flex flex-row"
```

Comment:
`border-[1px]` is the same as `border`, please use the already provided utility class rather than generating one